wcfsetup/install/files/lib/system/option/user/group/UserGroupOptionHandler.class.php

```diff
@@ -148,13 +166,17 @@ protected function isOwner() {
 	protected function validateOption(Option $option) {
 		parent::validateOption($option);
 		
-		if (!$this->isOwner()) {
-			// get type object
-			$typeObj = $this->getTypeObject($option->optionType);
-			
-			if ($typeObj->compare($this->optionValues[$option->optionName], WCF::getSession()->getPermission($option->optionName)) == 1) {
-				throw new UserInputException($option->optionName, 'exceedsOwnPermission');
-			}
+		if ($this->isOwner()) {
+			return;
+		}
+		
+		if (ENABLE_ENTERPRISE_MODE && $this->isAdmin() && !in_array($option->optionName, $this->enterpriseBlacklist)) {
```

Comment:
Should this condition be:
```suggestion
		if ($this->isAdmin() && (!ENABLE_ENTERPRISE_MODE || !in_array($option->optionName, $this->enterpriseBlacklist)) {
```

→ If the user is admin _and_ either the enterprise mode is disabled or the option is not blacklisted then the user may modify the option.

Reply:
(I used a separate commit, because you lacked a closing bracket and GitHub exploded or something when trying to edit the suggestion.)